src/components/app.js

```diff
@@ -41,10 +42,51 @@ const getMuiTheme = (theme) => {
     }
 };
 
-const App = () => {
+const SignInCallback = () => {
+    const history = useHistory();
+
+    function handleCallback() {
+        new UserManagerHelper().getUserManagerInstance()
+            .signinRedirectCallback().then(function () {
+            history.push("/");
+        }).catch(function (e) {
+            console.error(e);
+        });
+    }
+
+    useEffect(() => {
+        handleCallback();
+    }, []);
+
+    return (
+        <h1>Authentication callback processing...</h1>
+    )
+};
+
+const SilentRenew = () => {
```

Comment:
Can you explain the silentrenew flow ? What pages are open, what urls, when do they refresh, etc.
I'm not sure this correct (after all, it's a silent renew, and you are making a component to show "Silent renew in progress" ...
Maybe we can leave out silent renew for now ?